File: Core/DTOs/BorrowalsSearchDto.cs
using Core.Enums;

namespace Core.DTOs
{
    public class BorrowalsSearchDto
    {
        public string? BookName { get; set; }
        public List<int>? AuthorIds { get; set; }
        public List<BookGenre>? Genres { get; set; }
        public string? MemberName { get; set; }
        public string? MemberEmail { get; set; }
        public DateTime? BorrowedOn { get; set; }
        public DateTime? DueOn { get; set; }
        public List<BorrowalStatus>? Statuses { get; set; }
        public int? Delayed { get; set; }       
    }
}

File: client/src/app/shared/models/borrowals-search-dto.ts
import { BookGenre } from "./book-genre"
import { BorrowalStatus } from "./borrowal-status-enum"

export interface BorrowalsSearchDto {
    bookName?: string,
    authorIds?: number[],
    genres?: BookGenre[],
    memberName?: string,
    memberEmail?: string,
    borrowedOn?: Date,
    dueOn?: Date,
    statuses?: BorrowalStatus[],
    delayed?: number,
    applyFilters: boolean      
}